app/lib/account/backend.dart

```diff
@@ -131,42 +171,18 @@ Future<AuthenticatedUser> _requireAuthenticatedUser(
   return AuthenticatedUser(user, audience: auth.audience);
 }
 
-/// Require that the incoming request is authorized by an administrator with
-/// the given [permission].
-///
-/// Throws [AuthorizationException] if it doesn't have the permission.
-Future<AuthenticatedUser> requireAuthenticatedAdmin(
-    AdminPermission permission) async {
-  final authenticatedUser = await _requireAuthenticatedUser(
-      expectedAudience: activeConfiguration.adminAudience);
-  final user = authenticatedUser.user;
-  final isAdmin = await accountBackend.hasAdminPermission(
-    oauthUserId: authenticatedUser.oauthUserId,
-    email: authenticatedUser.email,
-    permission: permission,
-  );
-  if (!isAdmin) {
-    _logger.warning(
-        'User (${user.userId} / ${user.email}) is trying to access unauthorized admin APIs.');
-    throw AuthorizationException.userIsNotAdminForPubSite();
-  }
-  return authenticatedUser;
-}
-
-/// Verifies the current bearer token in the request scope and returns the
-/// current authenticated user or a service agent with the available data.
-Future<AuthenticatedAgent> requireAuthenticatedClient() async {
-  final token = _getBearerToken();
-  if (token == null || token.isEmpty) {
-    throw AuthenticationException.authenticationRequired();
+void _verifyAudience({
+  required String? expected,
```

Comment:
So that we can pass-through the `Configuration.xyzAudience` fields, and do the null-check inside this.
We could make those configuration values strictly non-nullable though...

Reply:
> We could make those configuration values strictly non-nullable though...

That might be safer... except our config files don't really support that, do they?
I don't mind crashing if we have misconfigured something -- crashing at start up is better than shipping a broken deployment.